src/domain/request/requestExecutor/RequestExecutorFactory.ts

```diff
@@ -0,0 +1,19 @@
+import InvalidContentTypeError from '@core/domain/common/utils/error/InvalidContentTypeError';
+import RequestExecutor from '@core/domain/common/port/RequestExecutorPort';
+import UrlRequestExecutor from './UrlRequestExecutor';
+
+class RequestExecutorFactory  {
+    create(contentType: string): RequestExecutor {
+        switch (contentType) {
```

Comment:
switch for 2 cases? I think `if`  will be better

Reply:
We weill very shortly have more cases there. This is basically what this refactor is about - we want to be able to select RequestExecutor by source.

Reply:
I'm not very in favor of switch statement with multiple cases providing the same result. How about creating `canHandle` boolean method in RequestsExecutor and then making a search in a collection of them.

Reply:
I actually wanted to d othis but forgot ;)